Ти керуєш інтернет-магазином, і тестуєш три різні варіанти відображення реклами для клієнтів (A, B, C). Метою є визначення, яка з них найбільш ефективна за кількістю кліків на рекламу (CTR — Click-Through Rate).

Проведи тест ANOVA на трьох різних версіях рекламної кампанії. Визнач, чи є значуща різниця між кампаніями.

**Опис даних:**
- `Variant`: Версія реклами (A, B або C).
- `Impressions`: Кількість показів реклами (скільки разів варіант реклами показувався користувачам).
- `Clicks`: Кількість кліків на рекламу (скільки користувачів клікнули на рекламу після її перегляду).
- `CTR (Click-Through Rate)`: Відношення кількості кліків до кількості показів, виражене у відсотках.

In [5]:
import pandas as pd

# Create the dataset
data = {
    "Variant": ["A", "A", "A", "B", "B", "B", "C", "C", "C", "A", "B", "C", "A", "B", "C"],
    "Impressions": [100, 150, 120, 90, 130, 140, 110, 100, 130, 160, 180, 120, 140, 110, 100],
    "Clicks": [10, 12, 14, 9, 15, 16, 11, 10, 14, 13, 17, 12, 15, 14, 13]
}

# Convert to DataFrame
df = pd.DataFrame(data)

# Add a column for CTR (Click-Through Rate)
df["CTR (%)"] = (df["Clicks"] / df["Impressions"]) * 100

# Display the dataset
print(df)

   Variant  Impressions  Clicks    CTR (%)
0        A          100      10  10.000000
1        A          150      12   8.000000
2        A          120      14  11.666667
3        B           90       9  10.000000
4        B          130      15  11.538462
5        B          140      16  11.428571
6        C          110      11  10.000000
7        C          100      10  10.000000
8        C          130      14  10.769231
9        A          160      13   8.125000
10       B          180      17   9.444444
11       C          120      12  10.000000
12       A          140      15  10.714286
13       B          110      14  12.727273
14       C          100      13  13.000000


Щоб виконати ANOVA-тест використовуючи дані з нашої таблиці, потрібно згрупувати дані за колонкою Variant і використати значення з колонки CTR (%)

In [11]:
import pandas as pd


ctr_a = df[df['Variant'] == 'A']['CTR (%)']
ctr_b = df[df['Variant'] == 'B']['CTR (%)']
ctr_c = df[df['Variant'] == 'C']['CTR (%)']


# Display the data
print(ctr_a)
print(ctr_b)
print(ctr_c)

0     10.000000
1      8.000000
2     11.666667
9      8.125000
12    10.714286
Name: CTR (%), dtype: float64
3     10.000000
4     11.538462
5     11.428571
10     9.444444
13    12.727273
Name: CTR (%), dtype: float64
6     10.000000
7     10.000000
8     10.769231
11    10.000000
14    13.000000
Name: CTR (%), dtype: float64


In [12]:
from scipy.stats import f_oneway


# Performing ANOVA test
f_stat, p_value = f_oneway(ctr_a, ctr_b, ctr_c)

print(f"F-statistic: {f_stat}")
print(f"P-value: {p_value}")

# Interpretation
if p_value < 0.05:
    print("There is a statistically significant difference between the campaigns.")
else:
    print("No statistically significant difference found between the campaigns.")

F-statistic: 1.2271941993395044
P-value: 0.32740790012546417
No statistically significant difference found between the campaigns.


**Висновок:**\
Оскільки p-value 0.3274 > 0.05, немає статистично значущої різниці між рекламними кампаніями A, B та C за показником CTR. Це означає, що за наявними даними жоден з варіантів реклами не показав суттєво вищої ефективності порівняно з іншими.